app/models/user.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  CONFIRMATION_TOKEN_EXPIRES_IN = 5.minutes
+
+  enum role: { admin: 0, doctor: 1, patient: 2 }
+  enum gender: { not_specified: 0, female: 1, male: 2 }
+  enum status: { pending: 0, active: 1, deleted: 2 }
+
+  after_create :generate_confirmation_token, if: -> { pending? }
+  after_create :send_confirmation_email, if: -> { pending? && confirmation_token.present? }
+
+  attr_accessor :confirmation_token
+
+  has_secure_password
+
+  validates :first_name, presence: true, length: { maximum: 128 }
+  validates :last_name, presence: true, length: { maximum: 128 }
+  validates :gender, :date_of_birth, :role, :password_digest, presence: true
```

Comment:
do not need to validate `:password_digest`

Reply:
@kth-wtag removed, apu.